### PR TITLE
WL-3326 : Hook up Lessons to Mneme (rather than Samigo)

### PR DESCRIPTION
**Pull request by nicholaswilson100**
4 main changes:

Enable the “optional” profile in tool/pom.xml
Remove the bits about Assignments 2 and Yaft in opt-applicationContext.xml
Set right version numbers and dependency type in main pom.xml
Change context.xmls to use MnemeEntity rather than SamigoEntity

### CODE REVIEW

File: tool/pom.xml [hunk 1]
```diff
@@ -256,6 +256,9 @@
  in shared/lib, so you want the scope to be "provided".
 -->
 
+        <activation>
```
Comment (buckett) [inline tool/pom.xml:259]: I'd prefer it if we weren't dependent on another locally run institutional maven repository, we've had problems in the past when local repos became unavailable.

**Comment by nicholaswilson100** [comment]
Yes that repository is actually unavailable now...https://sakai.rutgers.edu/stats/repos

**Comment by buckett** [comment]
If you're not using the rutgers repository how will the Yaft and Assignments2 dependencies be found?

**Comment by nicholaswilson100** [comment]
If we're not using Yaft or Assignment2, is there any reason i shouldn't remove these dependencies?

**Comment by buckett** [comment]
They can be dropped, but you'll also need to remove the code that compiles against them.

**Comment by nicholaswilson100** [comment]
ok

**Comment by nicholaswilson100** [comment]
thanks

**Comment by nicholaswilson100** [comment]
We don't need the yaft and assigment2 versions / scopes in the pom.xml either now

**Comment by buckett** [comment]
Would be really nice if this could be activated with a maven property as it would mean we weren't running a custom build of lessonbuilder, but I can't see how the requestContext.xml/applicationContext.xml changes could be profile activated.

**Comment by nicholaswilson100** [comment]
Yup

**Comment by nicholaswilson100** [comment]
Also, I'd really like to clean up the source in the tool in general.  Under the tool directory, there's /opt-src, /src, /src28, /src28-9 and /src29(!).  We surely don't need the 2.8, 2.9 stuff anymore do we  and could we put the opt-src into src?

**Pull request merged**